### PR TITLE
Update the KubernetesVersions in KubernetesVersion.java and in the places which use them

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Description

Updated the versions in Kubernetes.java and everywhere older versions were being used.
Solves the issue metioned in #4320 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java [hunk 1]
```diff
@@ -205,11 +205,8 @@ public boolean isOpenshift() {
     }
 
     public boolean isNamespaceAndPodSelectorNetworkPolicySupported() {
```
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java:207]: We should remove this method completely since it is only returning true now ... and also remove all its uses of course. We can do it in separate PR => but in that case we should open a new issue for it and link it here in a comment.

Comment (ShubhamRwt) [reply operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java:207]: Yes, I agree. I didn't removed it completely because it was used too widely in the code base.

File: operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java [hunk 2]
```diff
@@ -44,21 +44,21 @@ public static void after() {
...
         assertThat(pfa.isNamespaceAndPodSelectorNetworkPolicySupported(), is(true));
     }
 
     @Test
-    public void testVersionDetectionOpenShift39(VertxTestContext context) throws InterruptedException {
+    public void testVersionDetectionOpenShift44(VertxTestContext context) throws InterruptedException {
         String version = "{\n" +
                 "  \"major\": \"1\",\n" +
-                "  \"minor\": \"9\",\n" +
+                "  \"minor\": \"17\",\n" +
```
Comment (ShubhamRwt) [inline operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java:61]: Hello @scholzj,  Here do we require to change the git version also?

Comment (scholzj) [reply operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java:61]: Yeah, I think you should replace it here as well.

Comment (ShubhamRwt) [reply operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java:61]: I have changed the minor version and kubernetes version. Iam not sure changing `gitVersion\": \"v1.9.1+a0ce1bc657\" `.  Like can we can change it to any new version ?

Comment (scholzj) [reply operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java:61]: If you want you can use this from my cluster:

```
{
  "major": "1",
  "minor": "20",
  "gitVersion": "v1.20.1",
  "gitCommit": "c4d752765b3bbac2237bf87cf0b1c2e307844666",
  "gitTreeState": "clean",
  "buildDate": "2020-12-18T12:00:47Z",
  "goVersion": "go1.15.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

Comment (ShubhamRwt) [reply operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java]: @scholzj thanks for helping me out. I have updated the PR with the required tests.

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.
